pkg/kubelet/kubelet.go

```diff
@@ -654,6 +681,24 @@ func NewMainKubelet(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps *Kub
 
 	klet.statusManager = status.NewManager(klet.kubeClient, klet.podManager, klet)
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.RotateKubeletServerCertificate) {
+		names := append([]string{klet.GetHostname(), hostnameOverride}, cloudNames...)
+		klet.serverCertificateManager, err = initializeServerCertificateManager(klet.kubeClient, kubeCfg, klet.nodeName, cloudIPs, names)
+		if err != nil {
+			return nil, fmt.Errorf("failed to initialize certificate manager: %v", err)
+		}
+		if kubeDeps.TLSOptions != nil {
```

Comment:
I think TLS initialization else where. Certificate manager initialization happens here to get access to the client. I've shifted this check up to the top of the certificate manager initialization block so the certificate manager doesn't get initialized if it won't be used.